setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php

```diff
@@ -16,11 +16,25 @@ class ClassesScanner implements ClassesScannerInterface
     protected $excludePatterns = [];
 
     /**
+     * @var array
+     */
+
+    protected $fileResults = [];
+
+    /**
+     * @var DirectoryList
+     */
+
+    protected $directoryList;
+
+    /**
+     * @param DirectoryList $directoryList
      * @param array $excludePatterns
      */
-    public function __construct(array $excludePatterns = [])
+    public function __construct(DirectoryList $directoryList, array $excludePatterns = [])
```

Comment:
I wasn't aware of that practice, I have been trying like the plague to avoid using the ObjectManager like that.